WalletWasabi.Fluent/Models/Wallets/WalletModel.cs

```diff
@@ -108,22 +143,39 @@ public IAddress GetNextReceiveAddress(IEnumerable<string> destinationLabels)
 		return nextReceiveAddress;
 	}
 
-	public IWalletInfoModel GetWalletInfo()
+	private static string WalletFile(string walletDir, string walletWalletName) => Path.Combine(walletDir, walletWalletName + "." + WalletDirectories.WalletFileExtension);
+
+	private static void MoveFile(string sourceFileName, string destFileName)
 	{
-		return new WalletInfoModel(Wallet);
+		Logger.LogInfo($"Renaming file {sourceFileName} to {destFileName}");
+		File.Move(sourceFileName, destFileName);
 	}
 
-	public IWalletStatsModel GetWalletStats()
+	private void RenameWallet(string walletWalletName, string previousName)
 	{
-		return new WalletStatsModel(this, Wallet);
+		if (!IsValidWalletName(walletWalletName))
+		{
+			Logger.LogWarning($"Invalid name '{walletWalletName}' when attempting to rename '{previousName}'");
+			throw new InvalidOperationException($"Invalid name {walletWalletName}");
+		}
+
+		var walletDir = Services.WalletManager.WalletDirectories.WalletsDir;
+		MoveFile(WalletFile(walletDir, previousName), WalletFile(walletDir, walletWalletName));
+		try
+		{
+			var backupDir = Services.WalletManager.WalletDirectories.WalletsBackupDir;
+			MoveFile(WalletFile(backupDir, previousName), WalletFile(backupDir, walletWalletName));
+		}
+		catch (Exception e)
+		{
+			Logger.LogWarning($"Could not rename wallet backup file. Reason: {e.Message}");
+		}
+
+		Wallet.KeyManager.SetFilePath(WalletFile(walletDir, walletWalletName));
```

Comment:
The UI should not move files, instead it should call a `Rename` method in the "backend" code.

Reply:
I agree.